File: src/codelistings/txc1.cc
#include <omnetpp.h>

using namespace omnetpp;

class Txc1 : public cSimpleModule
{
    protected:
        virtual void initialize() override;
        virtual void handleMessage(cMessage *msg) override;
};

// The module class needs to be registered with OMNeT++
Define_Module(Txc1);

File: src/codelistings/txc1.py
from pyopp import cSimpleModule, cMessage

class PyTxc1(cSimpleModule):

    def initialize(self):
        if self.getName() == 'tic':
            msg = cMessage("pymessage")  # creación de mensaje nuevo
            self.send(msg, "out")

    def handleMessage(self, msg):
        self.send(msg, "out")
